components/feature/awesomebar/src/main/java/mozilla/components/feature/awesomebar/provider/HistoryMetadataSuggestionProvider.kt

```diff
@@ -27,12 +28,15 @@ internal const val METADATA_SUGGESTION_LIMIT = 5
  * for [HistoryMetadata] URLs.
  * @param engine optional [Engine] instance to call [Engine.speculativeConnect] for the
  * highest scored suggestion URL.
+ * @param maxNumberOfResults optional parameter allowing to lower the number of returned suggested
+ * history items to below the default of 5
  */
 class HistoryMetadataSuggestionProvider(
     private val historyStorage: HistoryMetadataStorage,
     private val loadUrlUseCase: SessionUseCases.LoadUrlUseCase,
     private val icons: BrowserIcons? = null,
-    internal val engine: Engine? = null
+    internal val engine: Engine? = null,
+    @VisibleForTesting internal val maxNumberOfResults: Int = -1
```

Comment:
The scenario I though about here is related to the filtering we do in both providers.
We ask for a certain limit of results and then possibly drop some items.
Seeing that the Fenix requirements are for a lower number of items than the limit I though of asking for that limit of items from which we'd be ok with filtering out some.
Otherwise, in those extreme cases in which items are filtered out Fenix could loose 1/3 or even more of the results wanted.
So I'd say there is a benefit but for only this particular case.
Not sure how often we'd filter out items. Can refactor this code if you don't think it helps.

Reply:
Yeah, don't feel strongly either way. In case the providers have additional filters, you're right that the results could be better if they fetched more and add the `take` after. So, feel free to keep it, would just increase the limit of the metadata provider then.